File: 226-invert-binary-tree/solution.cpp
#include <deque>

struct TreeNode
{
  int val;
  TreeNode *left;
  TreeNode *right;
  TreeNode() : val(0), left(nullptr), right(nullptr) {}
  TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
  TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution
{
public:
  TreeNode *invertTree(TreeNode *root)
  {
    std::deque<TreeNode *> queue{root};
    while (!queue.empty())
    {
      auto *node = queue.front();
      queue.pop_front();
      if (node == nullptr)
      {
        continue;
      }

      auto *left = node->left;
      auto *right = node->right;
      node->left = right;
      node->right = left;

      queue.emplace_back(left);
      queue.emplace_back(right);
    }

    return root;
  }
};

File: 226-invert-binary-tree/solution.py
from typing import Optional
from collections import deque


class TreeNode:
  def __init__(self, val=0, left=None, right=None):
    self.val = val
    self.left = left
    self.right = right


class Solution:
  def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
    queue: deque[Optional[TreeNode]] = deque([root])
    while queue:
      node = queue.popleft()
      if node is None:
        continue

      left, right = node.left, node.right
      node.left = right
      node.right = left

      queue.extend([left, right])

    return root
